src/routes/feedback/components/Feedback.js

```diff
@@ -0,0 +1,216 @@
+  title: string,
+  alias?: string,
+  query?: string,
+  route: string,
+  isPositiveRatingSelected: boolean,
+  pathname: string,
+  isOpen: boolean,
+  t: TFunction
+}
+
+type StateType = {
+  feedbackOptions: Array<FeedbackDropdownType>,
+  selectedFeedbackOption: FeedbackDropdownType,
+  comment: string
+}
+
+class Feedback extends React.Component<PropsType, StateType> {
+  constructor (props: PropsType) {
+    super(props)
+    const feedbackOptions = this.getFeedbackOptions()
+    this.state = {feedbackOptions: feedbackOptions, selectedFeedbackOption: feedbackOptions[0], comment: ''}
+  }
+
+  onCommentChanged = (event: {target: {value: string}}) => this.setState({comment: event.target.value})
+
+  onFeedbackOptionChanged = (selectedDropdown: FeedbackDropdownType) => {
+    this.setState({selectedFeedbackOption: selectedDropdown})
+  }
+
+  componentDidUpdate (prevProps: PropsType) {
+    const {isOpen} = this.props
+    const prevIsOpen = prevProps.isOpen
+
+    // If the Feedback is opened, we have to reset and initialize the state
+    if (prevIsOpen !== isOpen && isOpen) {
+      /* eslint-disable react/no-did-update-set-state */
+      const feedbackOptions = this.getFeedbackOptions()
+      const selectedFeedbackOption = feedbackOptions[0]
+      this.setState({feedbackOptions: feedbackOptions, selectedFeedbackOption: feedbackOptions[0], comment: ''})
+      FeedbackEndpoint.postData(this.getFeedbackData(selectedFeedbackOption, ''))
+    }
+  }
+
+  getFeedbackData = (selectedFeedbackOption: FeedbackDropdownType, comment: string): FeedbackDataType => {
+    const {id, city, language, alias, query, isPositiveRatingSelected} = this.props
+
+    return {
+      feedbackType: selectedFeedbackOption.feedbackType,
+      isPositiveRating: isPositiveRatingSelected,
+      comment,
+      id,
+      city: city || INTEGREAT_INSTANCE,
+      language: language || DEFAULT_FEEDBACK_LANGUAGE,
+      alias,
+      query
+    }
+  }
+
+  onSubmit = () => {
+    const {selectedFeedbackOption, comment} = this.state
+    FeedbackEndpoint.postData(this.getFeedbackData(selectedFeedbackOption, comment))
+  }
+
+  getFeedbackOptions = (): Array<FeedbackDropdownType> => {
+    const {cities, city, t} = this.props
+    const options = []
+    const currentPageFeedbackLabel = this.getCurrentPageFeedbackLabel()
+    if (currentPageFeedbackLabel) {
+      options.push(this.getFeedbackOption(currentPageFeedbackLabel, this.getCurrentPageFeedbackType()))
+    }
+    if (city) {
+      const cityTitle = CityModel.findCityName(cities, city)
+      options.push(this.getFeedbackOption(`${t('contentOfCity')} ${cityTitle}`, CATEGORIES_FEEDBACK_TYPE))
+    }
+
+    options.push(this.getFeedbackOption(t('technicalTopic'), CATEGORIES_FEEDBACK_TYPE))
+
+    return options
+  }
+
+  getFeedbackOption = (label: string, feedbackType: string | null): FeedbackDropdownType =>
+    ({value: label, feedbackType, label})
+
+  getCurrentPageFeedbackLabel = (): ?string => {
```

Comment:
You can refactor this component by extracting the components of the form into separate components.
Like a FeedbackDropdown, FeedbackInput.

That would make this component less complex.

Reply:
Thought about it, also did it, but imo it does not accomplish it (at least with FeedbackDropdown, because the dropdowns still have to live in the state of the Feedback(Box). Maybe it would work for FeedbackInput.